modules/tinymce/src/plugins/paste/test/ts/browser/ImagePasteTest.ts

```diff
@@ -57,214 +57,146 @@ UnitTest.asynctest('browser.tinymce.plugins.paste.ImagePasteTest', (success, fai
     return file;
   };
 
-  const mockEvent = (type: string, files) => {
-    const event = {
-      type,
-      preventDefault: Fun.noop
-    };
-
+  const mockEvent = <K extends 'drop' | 'paste'>(type: K, files: File[]): K extends 'drop' ? DragEvent : ClipboardEvent => {
     const transferName = type === 'drop' ? 'dataTransfer' : 'clipboardData';
-    event[transferName] = {
-      files
-    };
-
-    return event;
-  };
-
-  const setupContent = (editor: Editor) => {
-    editor.setContent('<p>a</p>');
-    LegacyUnit.setSelection(editor, 'p', 0);
-    return editor.selection.getRng();
+    return {
+      type,
+      preventDefault: Fun.noop,
+      [transferName]: { files }
+    } as any;
   };
 
-  const waitFor = (predicate: () => boolean): Promise<void> => {
-    return new Promise((resolve, reject) => {
-      const check = (time, count) => {
-        if (predicate()) {
-          resolve();
-        } else {
-          if (count === 0) {
-            reject(new Error('Waited for predicate to be true'));
-          } else {
-            Delay.setTimeout(() => {
-              check(time, count - 1);
-            }, time);
-          }
-        }
-      };
-
-      check(10, 100);
-    });
-  };
+  const pWaitFor = (predicate: () => boolean) =>
```

Comment:
Can we just kill this extra function now and call `Waiter.pTryUntilPredicate` directly in the test

Reply:
`pWaitFor` is used in a few cases and it removes the duplication with the label which is why I had decided to leave this.

Reply:
I'm with James on this one - not a strong opinion but on the side of simplifying e.g. because it's one layer I'd have to track through if this test failed and I was trying to figure out where

Reply:
That wouldn't be simplifying though? That'd be introducing more complexity.

Reply:
Looking at other PRs we've not inlined this and have kept the generic helper, so inlining this would be inconsistent (I spotted a number of similar cases in the search and replace one for example).

Reply:
Have removed it, but I still very much disagree with the reasons put forward. Neither are strong cases and goes against consistency/de-duplication efforts we'd normally apply.